app/components/new-sign-up/form.hbs

```diff
@@ -3,18 +3,16 @@
     {{this.label}}
   </h2>
   <div class="user-details__field">
-    <Input @value={{get @userDetails @state}} type="text" placeholder="Darth" {{on 'input' this.inputFieldChanged}} data-test-signup-form-input/>
+    <Input @value={{get @userDetails @currentStep}} type="text" placeholder="Darth" {{on 'input' this.inputFieldChanged}} data-test-signup-form-input/>
   </div>
 
   <div class="signup-action__btn">
-    <SignUp::Button
-      @clickHandler={{if (eq @state "username")
-      @register @changeRouteParams}}
-      @state={{@state}}
+    <NewSignUp::Button
+      @clickHandler={{@handleRouteParamsChange}}
       @disabled={{@isButtonDisabled}}
       @isSubmitClicked={{@isSubmitClicked}}
```

Comment:
Why does the button still need this prop?

Reply:
If this value is true, we show the spinner
<img width="610" alt="Screenshot 2022-08-15 at 10 11 23 AM" src="https://user-images.githubusercontent.com/29303618/184577977-fc13095e-e10d-487d-9728-178b1d870a68.png">